File: src/main/java/top/zbeboy/isy/service/InternshipApplyServiceImpl.java
package top.zbeboy.isy.service;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import org.jooq.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.ObjectUtils;
import org.springframework.util.StringUtils;
import top.zbeboy.isy.domain.tables.daos.InternshipApplyDao;
import top.zbeboy.isy.domain.tables.daos.InternshipCollegeDao;
import top.zbeboy.isy.domain.tables.pojos.InternshipApply;
import top.zbeboy.isy.domain.tables.pojos.Science;
import top.zbeboy.isy.service.util.DateTimeUtils;
import top.zbeboy.isy.service.util.SQLQueryUtils;
import top.zbeboy.isy.web.bean.internship.apply.InternshipApplyBean;
import top.zbeboy.isy.web.bean.internship.release.InternshipReleaseBean;
import top.zbeboy.isy.web.bean.internship.review.InternshipReviewBean;
import top.zbeboy.isy.web.util.PaginationUtils;

import javax.annotation.Resource;
import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import static top.zbeboy.isy.domain.Tables.*;

/**
 * Created by lenovo on 2016-11-29.
 */
@Service("internshipApplyService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
public class InternshipApplyServiceImpl implements InternshipApplyService {

    private final Logger log = LoggerFactory.getLogger(InternshipApplyServiceImpl.class);

    private final DSLContext create;

    private InternshipApplyDao internshipApplyDao;

    @Resource
    private InternshipReleaseScienceService internshipReleaseScienceService;

    @Autowired
    public InternshipApplyServiceImpl(DSLContext dslContext, Configuration configuration) {
        this.create = dslContext;
        this.internshipApplyDao = new InternshipApplyDao(configuration);
    }

    @Override
    public InternshipApply findById(String id) {
        return internshipApplyDao.findById(id);
    }

    @Override
    public Optional<Record> findByInternshipReleaseIdAndStudentId(String internshipReleaseId, int studentId) {
        return  create.select()
                .from(INTERNSHIP_APPLY)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.STUDENT_ID.eq(studentId)))
                .fetchOptional();
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    @Override
    public void save(InternshipApply internshipApply) {
        internshipApplyDao.insert(internshipApply);
    }

    @Override
    public void update(InternshipApply internshipApply) {
        internshipApplyDao.update(internshipApply);
    }

    @Override
    public void updateStateWithInternshipReleaseIdAndState(String internshipReleaseId, int changeState, int updateState) {
        create.update(INTERNSHIP_APPLY)
                .set(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE,updateState)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(changeState)))
                .execute();
    }

    @Override
    public void updateStateByChangeFillEndTime(Timestamp changeFillEndTime, int changeState, int updateState) {
        create.update(INTERNSHIP_APPLY)
                .set(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE,updateState)
                .where(INTERNSHIP_APPLY.CHANGE_FILL_END_TIME.le(changeFillEndTime).and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(changeState)))
                .execute();
    }

    @Override
    public void deleteByInternshipReleaseIdAndStudentId(String internshipReleaseId, int studentId) {
        create.deleteFrom(INTERNSHIP_APPLY)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.STUDENT_ID.eq(studentId)))
                .execute();
    }

    @Override
    public Result<Record> findAllByPage(PaginationUtils paginationUtils, InternshipApplyBean internshipApplyBean) {
        int pageNum = paginationUtils.getPageNum();
        int pageSize = paginationUtils.getPageSize();
        Condition a = searchCondition(paginationUtils);
        a = otherCondition(a, internshipApplyBean);
        return create.select()
                .from(INTERNSHIP_APPLY)
                .join(INTERNSHIP_RELEASE)
                .on(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_ID))
                .join(STUDENT)
                .on(INTERNSHIP_APPLY.STUDENT_ID.eq(STUDENT.STUDENT_ID))
                .join(INTERNSHIP_TYPE)
                .on(INTERNSHIP_TYPE.INTERNSHIP_TYPE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_TYPE_ID))
                .join(USERS)
                .on(INTERNSHIP_RELEASE.USERNAME.eq(USERS.USERNAME))
                .join(DEPARTMENT)
                .on(INTERNSHIP_RELEASE.DEPARTMENT_ID.eq(DEPARTMENT.DEPARTMENT_ID))
                .join(COLLEGE)
                .on(DEPARTMENT.COLLEGE_ID.eq(COLLEGE.COLLEGE_ID))
                .join(SCHOOL)
                .on(COLLEGE.COLLEGE_ID.eq(SCHOOL.SCHOOL_ID))
                .where(a)
                .orderBy(INTERNSHIP_APPLY.APPLY_TIME.desc())
                .limit((pageNum - 1) * pageSize, pageSize)
                .fetch();
    }

    @Override
    public List<InternshipApplyBean> dealData(PaginationUtils paginationUtils, Result<Record> records, InternshipApplyBean internshipApplyBean) {
        List<InternshipApplyBean> internshipApplyBeens = new ArrayList<>();
        if (records.isNotEmpty()) {
            internshipApplyBeens = records.into(InternshipApplyBean.class);
            String format = "yyyy-MM-dd HH:mm:ss";
            internshipApplyBeens.forEach(i -> {
                i.setTeacherDistributionStartTimeStr(DateTimeUtils.timestampToString(i.getTeacherDistributionStartTime(), format));
                i.setTeacherDistributionEndTimeStr(DateTimeUtils.timestampToString(i.getTeacherDistributionEndTime(), format));
                i.setStartTimeStr(DateTimeUtils.timestampToString(i.getStartTime(), format));
                i.setEndTimeStr(DateTimeUtils.timestampToString(i.getEndTime(), format));
                i.setReleaseTimeStr(DateTimeUtils.timestampToString(i.getReleaseTime(), format));
                Result<Record> records1 = internshipReleaseScienceService.findByInternshipReleaseIdRelation(i.getInternshipReleaseId());
                i.setSciences(records1.into(Science.class));
            });
            paginationUtils.setTotalDatas(countByCondition(paginationUtils, internshipApplyBean));
        }
        return internshipApplyBeens;
    }

    @Override
    public int countByCondition(PaginationUtils paginationUtils, InternshipApplyBean internshipApplyBean) {
        Record1<Integer> count;
        Condition a = searchCondition(paginationUtils);
        a = otherCondition(a, internshipApplyBean);
        if (ObjectUtils.isEmpty(a)) {
            SelectJoinStep<Record1<Integer>> selectJoinStep = create.selectCount()
                    .from(INTERNSHIP_APPLY);
            count = selectJoinStep.fetchOne();
        } else {
            SelectConditionStep<Record1<Integer>> selectConditionStep = create.selectCount()
                    .from(INTERNSHIP_APPLY)
                    .join(INTERNSHIP_RELEASE)
                    .on(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_ID))
                    .join(STUDENT)
                    .on(INTERNSHIP_APPLY.STUDENT_ID.eq(STUDENT.STUDENT_ID))
                    .join(INTERNSHIP_TYPE)
                    .on(INTERNSHIP_TYPE.INTERNSHIP_TYPE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_TYPE_ID))
                    .join(USERS)
                    .on(INTERNSHIP_RELEASE.USERNAME.eq(USERS.USERNAME))
                    .join(DEPARTMENT)
                    .on(INTERNSHIP_RELEASE.DEPARTMENT_ID.eq(DEPARTMENT.DEPARTMENT_ID))
                    .join(COLLEGE)
                    .on(DEPARTMENT.COLLEGE_ID.eq(COLLEGE.COLLEGE_ID))
                    .join(SCHOOL)
                    .on(COLLEGE.COLLEGE_ID.eq(SCHOOL.SCHOOL_ID))
                    .where(a);
            count = selectConditionStep.fetchOne();
        }
        return count.value1();
    }

    /**
     * 搜索条件
     *
     * @param paginationUtils 分页工具
     * @return 条件
     */
    public Condition searchCondition(PaginationUtils paginationUtils) {
        Condition a = null;
        JSONObject search = JSON.parseObject(paginationUtils.getSearchParams());
        if (!ObjectUtils.isEmpty(search)) {
            String internshipTitle = StringUtils.trimWhitespace(search.getString("internshipTitle"));
            String internshipReleaseId = StringUtils.trimWhitespace(search.getString("internshipReleaseId"));
            if (StringUtils.hasLength(internshipTitle)) {
                a = INTERNSHIP_RELEASE.INTERNSHIP_TITLE.like(SQLQueryUtils.likeAllParam(internshipTitle));
            }

            if (StringUtils.hasLength(internshipReleaseId)) {
                if(!ObjectUtils.isEmpty(a)){
                    a = a.and(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId));
                } else {
                    a = INTERNSHIP_RELEASE.INTERNSHIP_TITLE.like(SQLQueryUtils.likeAllParam(internshipTitle));
                }
            }
        }
        return a;
    }

    /**
     * 其它条件参数
     *
     * @param a                 搜索条件
     * @param internshipApplyBean 额外参数
     * @return 条件
     */
    private Condition otherCondition(Condition a, InternshipApplyBean internshipApplyBean) {
        if (!ObjectUtils.isEmpty(internshipApplyBean)) {
            if (!ObjectUtils.isEmpty(internshipApplyBean.getStudentId()) && internshipApplyBean.getStudentId()>0) {
                if (!ObjectUtils.isEmpty(a)) {
                    a = a.and(INTERNSHIP_APPLY.STUDENT_ID.eq(internshipApplyBean.getStudentId()));
                } else {
                    a = INTERNSHIP_APPLY.STUDENT_ID.eq(internshipApplyBean.getStudentId());
                }
            }

            if (StringUtils.hasLength(internshipApplyBean.getInternshipReleaseId())) {
                if (!ObjectUtils.isEmpty(a)) {
                    a = a.and(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipApplyBean.getInternshipReleaseId()));
                } else {
                    a = INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipApplyBean.getInternshipReleaseId());
                }
            }

            if (!ObjectUtils.isEmpty(internshipApplyBean.getInternshipReleaseIsDel())) {
                if (!ObjectUtils.isEmpty(a)) {
                    a = a.and(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_IS_DEL.eq(internshipApplyBean.getInternshipReleaseIsDel()));
                } else {
                    a = INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_IS_DEL.eq(internshipApplyBean.getInternshipReleaseIsDel());
                }
            }

            if (!ObjectUtils.isEmpty(internshipApplyBean.getInternshipApplyState())) {
                if (!ObjectUtils.isEmpty(a)) {
                    a = a.and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(internshipApplyBean.getInternshipApplyState()));
                } else {
                    a = INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(internshipApplyBean.getInternshipApplyState());
                }
            }
        }
        return a;
    }
}

File: src/main/java/top/zbeboy/isy/service/internship/InternshipApplyServiceImpl.kt
package top.zbeboy.isy.service.internship

import com.alibaba.fastjson.JSON
import org.jooq.*
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional
import org.springframework.util.ObjectUtils
import org.springframework.util.StringUtils
import top.zbeboy.isy.config.Workbook

import top.zbeboy.isy.domain.Tables.*
import top.zbeboy.isy.domain.tables.daos.InternshipApplyDao
import top.zbeboy.isy.domain.tables.pojos.InternshipApply
import top.zbeboy.isy.domain.tables.pojos.Science
import top.zbeboy.isy.domain.tables.records.InternshipApplyRecord
import top.zbeboy.isy.service.util.DateTimeUtils
import top.zbeboy.isy.service.util.SQLQueryUtils
import top.zbeboy.isy.web.bean.internship.apply.InternshipApplyBean
import top.zbeboy.isy.web.util.PaginationUtils
import java.sql.Timestamp
import java.util.*
import javax.annotation.Resource

/**
 * Created by zbeboy 2017-12-27 .
 **/
@Service("internshipApplyService")
@Transactional(propagation = Propagation.SUPPORTS, readOnly = true)
open class InternshipApplyServiceImpl @Autowired constructor(dslContext: DSLContext) : InternshipApplyService {

    private val create: DSLContext = dslContext

    @Resource
    open lateinit var internshipApplyDao: InternshipApplyDao

    @Resource
    open lateinit var internshipReleaseScienceService: InternshipReleaseScienceService

    @Resource
    open lateinit var internshipTypeService: InternshipTypeService

    @Resource
    open lateinit var internshipCollegeService: InternshipCollegeService

    @Resource
    open lateinit var internshipCompanyService: InternshipCompanyService

    @Resource
    open lateinit var graduationPracticeCollegeService: GraduationPracticeCollegeService

    @Resource
    open lateinit var graduationPracticeCompanyService: GraduationPracticeCompanyService

    @Resource
    open lateinit var graduationPracticeUnifyService: GraduationPracticeUnifyService


    override fun findById(id: String): InternshipApply {
        return internshipApplyDao.findById(id)
    }

    override fun findByInternshipReleaseIdAndStudentId(internshipReleaseId: String, studentId: Int): Optional<Record> {
        return create.select()
                .from(INTERNSHIP_APPLY)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.STUDENT_ID.eq(studentId)))
                .fetchOptional()
    }

    @Transactional(propagation = Propagation.REQUIRED, readOnly = false)
    override fun save(internshipApply: InternshipApply) {
        internshipApplyDao.insert(internshipApply)
    }

    override fun update(internshipApply: InternshipApply) {
        internshipApplyDao.update(internshipApply)
    }

    override fun updateStateWithInternshipReleaseIdAndState(internshipReleaseId: String, changeState: Int, updateState: Int) {
        create.update<InternshipApplyRecord>(INTERNSHIP_APPLY)
                .set(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE, updateState)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(changeState)))
                .execute()
    }

    override fun updateStateByChangeFillEndTime(changeFillEndTime: Timestamp, changeState: Int, updateState: Int) {
        create.update<InternshipApplyRecord>(INTERNSHIP_APPLY)
                .set(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE, updateState)
                .where(INTERNSHIP_APPLY.CHANGE_FILL_END_TIME.le(changeFillEndTime).and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(changeState)))
                .execute()
    }

    override fun deleteByInternshipReleaseIdAndStudentId(internshipReleaseId: String, studentId: Int) {
        create.deleteFrom<InternshipApplyRecord>(INTERNSHIP_APPLY)
                .where(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipReleaseId).and(INTERNSHIP_APPLY.STUDENT_ID.eq(studentId)))
                .execute()
    }

    override fun deleteInternshipApplyRecord(internshipTypeId: Int, internshipReleaseId: String, studentId: Int) {
        val internshipType = internshipTypeService.findByInternshipTypeId(internshipTypeId)
        when (internshipType.internshipTypeName) {
            Workbook.INTERNSHIP_COLLEGE_TYPE -> internshipCollegeService.deleteByInternshipReleaseIdAndStudentId(internshipReleaseId, studentId)
            Workbook.INTERNSHIP_COMPANY_TYPE -> internshipCompanyService.deleteByInternshipReleaseIdAndStudentId(internshipReleaseId, studentId)
            Workbook.GRADUATION_PRACTICE_COLLEGE_TYPE -> graduationPracticeCollegeService.deleteByInternshipReleaseIdAndStudentId(internshipReleaseId, studentId)
            Workbook.GRADUATION_PRACTICE_UNIFY_TYPE -> graduationPracticeUnifyService.deleteByInternshipReleaseIdAndStudentId(internshipReleaseId, studentId)
            Workbook.GRADUATION_PRACTICE_COMPANY_TYPE -> graduationPracticeCompanyService.deleteByInternshipReleaseIdAndStudentId(internshipReleaseId, studentId)
        }
    }

    override fun findAllByPage(paginationUtils: PaginationUtils, internshipApplyBean: InternshipApplyBean): Result<Record> {
        val pageNum = paginationUtils.getPageNum()
        val pageSize = paginationUtils.getPageSize()
        var a = searchCondition(paginationUtils)
        a = otherCondition(a, internshipApplyBean)
        return create.select()
                .from(INTERNSHIP_APPLY)
                .join(INTERNSHIP_RELEASE)
                .on(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_ID))
                .join(STUDENT)
                .on(INTERNSHIP_APPLY.STUDENT_ID.eq(STUDENT.STUDENT_ID))
                .join(INTERNSHIP_TYPE)
                .on(INTERNSHIP_TYPE.INTERNSHIP_TYPE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_TYPE_ID))
                .join(DEPARTMENT)
                .on(INTERNSHIP_RELEASE.DEPARTMENT_ID.eq(DEPARTMENT.DEPARTMENT_ID))
                .join(COLLEGE)
                .on(DEPARTMENT.COLLEGE_ID.eq(COLLEGE.COLLEGE_ID))
                .join(SCHOOL)
                .on(COLLEGE.COLLEGE_ID.eq(SCHOOL.SCHOOL_ID))
                .leftJoin(FILES)
                .on(INTERNSHIP_APPLY.INTERNSHIP_FILE_ID.eq(FILES.FILE_ID))
                .where(a)
                .orderBy(INTERNSHIP_APPLY.APPLY_TIME.desc())
                .limit((pageNum - 1) * pageSize, pageSize)
                .fetch()
    }

    override fun dealData(paginationUtils: PaginationUtils, records: Result<Record>, internshipApplyBean: InternshipApplyBean): List<InternshipApplyBean> {
        var internshipApplyBeens: List<InternshipApplyBean> = ArrayList()
        if (records.isNotEmpty) {
            internshipApplyBeens = records.into(InternshipApplyBean::class.java)
            val format = "yyyy-MM-dd HH:mm:ss"
            internshipApplyBeens.forEach { i ->
                i.teacherDistributionStartTimeStr = DateTimeUtils.timestampToString(i.teacherDistributionStartTime!!, format)
                i.teacherDistributionEndTimeStr = DateTimeUtils.timestampToString(i.teacherDistributionEndTime!!, format)
                i.startTimeStr = DateTimeUtils.timestampToString(i.startTime!!, format)
                i.endTimeStr = DateTimeUtils.timestampToString(i.endTime!!, format)
                i.releaseTimeStr = DateTimeUtils.timestampToString(i.releaseTime!!, format)
                val records1 = internshipReleaseScienceService.findByInternshipReleaseIdRelation(i.internshipReleaseId)
                i.sciences = records1.into(Science::class.java)
            }
            paginationUtils.setTotalDatas(countByCondition(paginationUtils, internshipApplyBean))
        }
        return internshipApplyBeens
    }

    override fun countByCondition(paginationUtils: PaginationUtils, internshipApplyBean: InternshipApplyBean): Int {
        val count: Record1<Int>
        var a = searchCondition(paginationUtils)
        a = otherCondition(a, internshipApplyBean)
        if (ObjectUtils.isEmpty(a)) {
            val selectJoinStep = create.selectCount()
                    .from(INTERNSHIP_APPLY)
            count = selectJoinStep.fetchOne()
        } else {
            val selectConditionStep = create.selectCount()
                    .from(INTERNSHIP_APPLY)
                    .join(INTERNSHIP_RELEASE)
                    .on(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_ID))
                    .join(STUDENT)
                    .on(INTERNSHIP_APPLY.STUDENT_ID.eq(STUDENT.STUDENT_ID))
                    .join(INTERNSHIP_TYPE)
                    .on(INTERNSHIP_TYPE.INTERNSHIP_TYPE_ID.eq(INTERNSHIP_RELEASE.INTERNSHIP_TYPE_ID))
                    .join(DEPARTMENT)
                    .on(INTERNSHIP_RELEASE.DEPARTMENT_ID.eq(DEPARTMENT.DEPARTMENT_ID))
                    .join(COLLEGE)
                    .on(DEPARTMENT.COLLEGE_ID.eq(COLLEGE.COLLEGE_ID))
                    .join(SCHOOL)
                    .on(COLLEGE.COLLEGE_ID.eq(SCHOOL.SCHOOL_ID))
                    .where(a)
            count = selectConditionStep.fetchOne()
        }
        return count.value1()
    }

    /**
     * 搜索条件
     *
     * @param paginationUtils 分页工具
     * @return 条件
     */
    fun searchCondition(paginationUtils: PaginationUtils): Condition? {
        var a: Condition? = null
        val search = JSON.parseObject(paginationUtils.getSearchParams())
        if (!ObjectUtils.isEmpty(search)) {
            val internshipTitle = StringUtils.trimWhitespace(search.getString("internshipTitle"))
            if (StringUtils.hasLength(internshipTitle)) {
                a = INTERNSHIP_RELEASE.INTERNSHIP_TITLE.like(SQLQueryUtils.likeAllParam(internshipTitle))
            }
        }
        return a
    }

    /**
     * 其它条件参数
     *
     * @param a                   搜索条件
     * @param internshipApplyBean 额外参数
     * @return 条件
     */
    private fun otherCondition(a: Condition?, internshipApplyBean: InternshipApplyBean): Condition? {
        var tempCondition = a
        if (!ObjectUtils.isEmpty(internshipApplyBean)) {
            if (!ObjectUtils.isEmpty(internshipApplyBean.studentId) && internshipApplyBean.studentId > 0) {
                tempCondition = if (!ObjectUtils.isEmpty(tempCondition)) {
                    tempCondition!!.and(INTERNSHIP_APPLY.STUDENT_ID.eq(internshipApplyBean.studentId))
                } else {
                    INTERNSHIP_APPLY.STUDENT_ID.eq(internshipApplyBean.studentId)
                }
            }

            if (StringUtils.hasLength(internshipApplyBean.internshipReleaseId)) {
                tempCondition = if (!ObjectUtils.isEmpty(tempCondition)) {
                    tempCondition!!.and(INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipApplyBean.internshipReleaseId))
                } else {
                    INTERNSHIP_APPLY.INTERNSHIP_RELEASE_ID.eq(internshipApplyBean.internshipReleaseId)
                }
            }

            if (!ObjectUtils.isEmpty(internshipApplyBean.internshipReleaseIsDel)) {
                tempCondition = if (!ObjectUtils.isEmpty(tempCondition)) {
                    tempCondition!!.and(INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_IS_DEL.eq(internshipApplyBean.internshipReleaseIsDel))
                } else {
                    INTERNSHIP_RELEASE.INTERNSHIP_RELEASE_IS_DEL.eq(internshipApplyBean.internshipReleaseIsDel)
                }
            }

            if (!ObjectUtils.isEmpty(internshipApplyBean.internshipApplyState)) {
                tempCondition = if (!ObjectUtils.isEmpty(tempCondition)) {
                    tempCondition!!.and(INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(internshipApplyBean.internshipApplyState))
                } else {
                    INTERNSHIP_APPLY.INTERNSHIP_APPLY_STATE.eq(internshipApplyBean.internshipApplyState)
                }
            }
        }
        return tempCondition
    }

}